src/components/schedules/sidebar/Modules.jsx

```diff
@@ -2,71 +2,170 @@ import React from "react";
 import Nav from "react-bootstrap/Nav";
 import { scroller } from "react-scroll";
 
-// not being used at the moment
-function Modules({ scheduleData }) {
-  const moduleArray = [];
+function Modules({ scheduleData, coursetype }) {
+  const moduleNameArray = [];
+  const scheduleUrls = [];
+  const scheduleObjs = [];
 
-  // getting names of all modules
   Object.keys(scheduleData).map((day) => {
     if (scheduleData[day].dateTypes.module) {
-      if (!moduleArray.includes(scheduleData[day].dateTypes.module)) {
-        moduleArray.push(scheduleData[day].dateTypes.module);
-      }
+      Object.keys(scheduleData[day].dateTypes)
+      .filter(section => section !== 'module')
+      .map((section) => {
+        Object.keys(scheduleData[day].dateTypes[section])
+        .filter(classtype => classtype !== 'type')
+        .map((classtype) => {
+          if (scheduleData[day].dateTypes[section][classtype].items) {
+            scheduleData[day].dateTypes[section][classtype].items.map((item) => {
+              if(item.url && !scheduleUrls.includes(item.url)) {
+                scheduleUrls.push(item.url);
+                scheduleObjs.push({name: item.name, url: item.url, date: day});
+              }
+              if (item.url) {
+                const itemUrlArray = item.url.split('/');
+                if (!moduleNameArray.includes(itemUrlArray[3]) && itemUrlArray[2] === 'bootcamp.rocketacademy.co') {
```

Comment:
using 3 here: `itemUrlArray[3]` is not a good practice bc I'm not sure what 3 means- 

ideally  I think there would be:

1. a comment explaining what 3 is
2. `itemUrlArray[3]` should be made into a named variable